tls/ttls.c

```diff
@@ -141,6 +141,9 @@ ttls_crypto_req_sglist(TlsCtx *tls, struct crypto_aead *tfm, unsigned int len,
 	unsigned int sz, aead_sz, to_read, off;
 	int n;
 
+	WARN_ON_ONCE(len == 0); /* nothing to decrypt */
+	BUG_ON(!buf && !skb);
```

Comment:
As @ikoveshnikov said, it shouldn't cause a crash, so WARN_ON or WARN_ON_ONCE should be enough. I still like to keep that as a warning, rather than removing completely, as this condition means something went wrong at the caller.